spec/electric_slide/call_queue_spec.rb

```diff
@@ -138,6 +145,34 @@
           end
         end
       end
+
+      context "when the agent's and caller's calls are joined" do
+        before do
+          queued_call << Punchblock::Event::Joined.new(timestamp: connected_time)
+          agent_call << Punchblock::Event::Joined.new(timestamp: connected_time)
+        end
+
+        it "records the connection time in the :electric_slide_connected_at call variable on the queued call" do
+          expect(queued_call[:electric_slide_connected_at]).to eq(connected_time)
+        end
+      end
+
+      context "when the agent's and caller's calls are not joined" do
+        context 'and the call ends' do
+          before do
+            # prevent the agent from being returned to the queue so the queued
+            # call isn't grabbed by the agent again, changing queued call state
+            # before the example can check it
+            allow(agent).to receive(:presence).and_return(:unavailable)
```

Comment:
You mean via `#remove_agent`? On hanging up, [`#conditionally_return_agent`](https://github.com/neildecapia/electric_slide/blob/feature/queued-call-agent-variable/lib/electric_slide/call_queue.rb#L316) will [raise a `MissingAgentError`](https://github.com/neildecapia/electric_slide/blob/feature/queued-call-agent-variable/lib/electric_slide/call_queue.rb#L124) and cause [the rest of the call end handler](https://github.com/neildecapia/electric_slide/blob/feature/queued-call-agent-variable/lib/electric_slide/call_queue.rb#L318-L331) -- including the unsetting of the `:agent` call variable -- to go uncalled.